src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,92 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, deleteComment } from '../../api/comments';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ selectedPostId }) => {
+  const [post, setPost] = useState('');
+  const [comments, setComments] = useState('');
+  const [isCommentVisible, setIsCommentVisible] = useState(false);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(result => setPost(result));
+    getPostComments(selectedPostId)
+      .then(result => setComments(result));
+  }, [selectedPostId]);
+
+  const handleButton = () => {
+    setIsCommentVisible(!isCommentVisible);
+  };
+
+  const handleDeleteComment = (commentId) => {
+    deleteComment(commentId);
+
+    setComments(prevComments => prevComments.filter(
+      comment => comment.id !== commentId,
+    ));
+  };
+
+  const addComment = (comment) => {
+    setComments(prevComments => [...prevComments, comment]);
+  };
```

Comment:
you need to make a new request to the server and fetch a new list of comments to omit different states on the client and server

`add/remove something on the client -> make a new request to the server and fetch the result `